### ISSUE TITLE
Implement comment pinning.

### ISSUE THREAD

**Issue by RokeJulianLockhart** [mastodon/mastodon#27517]
### Pitch

Allow OPs and moderators to pin comments on posts.

### Motivation

https://github.com/mastodon/mastodon/issues/27302#issuecomment-1775260636

**Comment by RokeJulianLockhart** [comment]
We reuse the code used for https://allthings.how/how-to-pin-a-post-toot-on-mastodon/ erhaps.

**Comment by brendanjones** [comment]
@RokeJulianLockhart  can you give some detail about exactly what you mean? Do you mean the ability to, from all existing replies to a post, pin one of them at the top so it appears above all others, out of chronological order?

Could be useful, like if the OP posts a question and they want to pin the best answer. 

Would the pinned comment, if it’s not by the OP, appear above any replies by the OP?

**Comment by JustDalek** [comment]
Love this, it would also be nice if pinned comments worked like pinned posts on a profile where people who do not normally federate with that reply get to see it!

**Comment by RockstarRaccoon** [comment]
> it would also be nice if pinned comments worked like pinned posts on a profile where people who do not normally federate with that reply get to see it!

Yeah, I think this would be part of the proposed standard, that it gets passed along.  If you look at the original proposal, the main use-case is for mods of a server to toss down fact-check-type notes for posts from that server.  I just rolled back to it because apparently Timnit Gebru is dealing with reports from people who think talking about the Israeli-Palestinian-conflict violates European laws: they do not, and a pinned post with proof would hopefully get her a few less reports.  I would also appreciate being able to comment, as a moderator, either a fact check or simply a note about "We saw this post, this is where the line is, this is why we didn't take this down."

**Comment by RokeJulianLockhart** [comment]
> https://github.com/mastodon/mastodon/issues/27517#issuecomment-1779464574

@brendanjones, yes, I refer to what you describe.
